### DISCUSSION TITLE
Samples

### DISCUSSION THREAD

**Discussion by vbjay** [skoruba/IdentityServer4.Admin#920] [Show and tell]
I have created a sample repo that provides not  only having your IDS4 admin and STS, but also built some sample clients and configured them to work.  The repo is located [here](https://github.com/vbjay/IDS4-Sample) .  In the repo you will also find a net 5 winforms client that you can run and authenticate against the debugged version in the repo.  

Live site urls:
- [Sample Oidc client App](https://demoids.vbjaysolutions.com/mvc)
- [Sample Weather API](https://demoids.vbjaysolutions.com/weather/swagger)
- [Identity Server](https://demoids.vbjaysolutions.com/sts)
- [Identity Server Admin](https://demoids.vbjaysolutions.com/admin)
- [Identity Server Admin API](https://demoids.vbjaysolutions.com/adminui.api/swagger)

You can gain access by logging in with a Google or Github account.  It will give you a readonly admin role by default that will allow you to view the configuration for the items but you do not have access to modify things.  

I do need to upgrade the repo to use the latest template instead of the beta along with getting npm packages up to date and alter my deployment to use more current node/npm.  However, if this helps in going from you have an ids4 and seeing how to set up clients to make things work...then I have reached my goal.

Hope it helps.

**Comment by skoruba** [comment]
Hey @vbjay - thank you for providing these samples, I will check it. 👍

**Comment by skoruba** [comment]
BTW: @vbjay - I can see that you have nicely built automatic deployment - where do you host your sites? 
I would like to setup some sample deployment as well - help is appreciated. 😊 Thanks

**Reply by vbjay** [reply]
https://ci.appveyor.com/project/vbjay/ids4-sample is where it builds and deploys from. 

I use smarterasp.net for my hosting.

**Reply by vbjay** [reply]
If you need help getting it going, we can do a call sometime.

**Reply by skoruba** [reply]
Can you please mail me your contact? Thx

**Comment by vbjay** [comment]
I  am working on FIDO2 support.  
![image](https://user-images.githubusercontent.com/3629489/145515591-2cdaa0c7-935f-405c-b2d3-2414b1632799.png)
![image](https://user-images.githubusercontent.com/3629489/145515638-1ae7c18e-47f9-4e32-9bb5-9b2b516fcf84.png)
![image](https://user-images.githubusercontent.com/3629489/145515719-87b2a65c-8fc1-4348-8df3-c48f3827002c.png)
![image](https://user-images.githubusercontent.com/3629489/145516095-553a63a8-0c15-4e1f-938f-273f42fab646.png)
![image](https://user-images.githubusercontent.com/3629489/145516128-e62ed21c-6694-432b-96c6-0421240cd0f5.png)
https://www.pcmag.com/reviews/yubico-yubikey-5c-nfc is what I bought as  a security key and google supports using your phone as a key and getting tap this to authenticate from your phone in chrome.

**Reply by skoruba** [reply]
Nice 👍

**Reply by vbjay** [reply]
Check it out.   https://demoids.vbjaysolutions.com/sts  It is live with FIDO2 support.

![image](https://user-images.githubusercontent.com/3629489/145752370-cfd2b97a-4ae4-4a5e-8e9c-827015d788fb.png)

**Reply by skoruba** [reply]
Nice 👍 thank you for sharing this feature. 😊👏

**Comment by desarrolloce** [comment]
Hello "muchachos".  Greetings from Argentina.

I'm interested in the work of adding FIDO2 into Skoruba's UI.

Exactly wich code must add and where??

There is some repo from i can download code??

**Reply by vbjay** [reply]
![Screenshot_20230704-063118_GitHub.jpg](https://github.com/skoruba/IdentityServer4.Admin/assets/3629489/88a9dc02-4f8d-4fca-85b6-51d9a68621ae)

**Reply by desarrolloce** [reply]
Hello VBJAY.  Thanks for you response.

I'v allready use the repo you say:  [https://github.com/vbjay/IDS4-Sample](url) 

But i dont see any reference to FIDO2.

When i run this code in Visual Studio, obtain:

![image](https://github.com/skoruba/IdentityServer4.Admin/assets/63314874/b4fa1098-e746-4bee-94cf-860b365cd2b0)


Versus you test site: [https://demoids.vbjaysolutions.com/sts](url)

![image](https://github.com/skoruba/IdentityServer4.Admin/assets/63314874/968f04ba-3ee9-4a8a-a025-7a0a72ab81d4)

maybe i'm doing something wrong

**Reply by vbjay** [reply]
Look at all the branches.  I ci/cd deploy develop to demoids site. I deploy master to a different site.

**Comment by desarrolloce** [comment]
YES MI FRIEND !!!!!!

It's rigth here: [https://github.com/vbjay/IDS4-Sample/tree/fido](url)

of course i'll continue looking in subsecuents branchs

sorry for bothering you. Thanks a lot.
